src/data/species-form-change-triggers/meloetta-form-change-post-move-trigger.ts

```diff
@@ -0,0 +1,19 @@
+import { SpeciesFormChangePostMoveTrigger } from "#app/data/species-form-change-triggers/species-form-change-post-move-trigger";
+import type { Pokemon } from "#app/field/pokemon";
+import { globalScene } from "#app/global-scene";
+import { Abilities } from "#enums/abilities";
+import { Challenges } from "#enums/challenges";
+
+export class MeloettaFormChangePostMoveTrigger extends SpeciesFormChangePostMoveTrigger {
+  override canChange(pokemon: Pokemon): boolean {
+    if (globalScene.gameMode.hasChallenge(Challenges.SINGLE_TYPE)) {
```

Comment:
```suggestion
    // TODO: improve this (should only block the form change in Psychic or Fighting mono-type, not Normal)
    if (globalScene.gameMode.hasChallenge(Challenges.SINGLE_TYPE)) {
```
Something I forgot about when implementing this.